app/controllers/v1/sessions_controller.rb

```diff
@@ -138,23 +138,49 @@ def user_login(saml_response)
     end
 
     def render_login(type)
-      login_url, post_params = login_params(type)
+      login_url, @post_params = login_params(type)
       tracker = url_service.tracker
       renderer = ActionController::Base.renderer
       renderer.controller.prepend_view_path(Rails.root.join('lib', 'saml', 'templates'))
       result = renderer.render template: 'sso_post_form',
                                locals: {
                                  url: login_url,
-                                 params: post_params,
+                                 params: @post_params,
                                  id: tracker.uuid,
                                  authn: tracker.payload_attr(:authn_context),
                                  type: tracker.payload_attr(:type)
                                },
                                format: :html
       render body: result, content_type: 'text/html'
+      set_sso_saml_cookie!
       saml_request_stats
     end
 
+    def set_sso_saml_cookie!
+      cookies[Settings.ssoe_eauth_cookie.name] = {
+        value: saml_cookie_content.to_json,
+        expires: nil,
+        secure: Settings.ssoe_eauth_cookie.secure,
+        httponly: true,
+        domain: Settings.ssoe_eauth_cookie.domain
+      }
+    end
+
+    def saml_cookie_content
+      transaction_id = if current_user && url_service.should_uplevel?
```

Comment:
cool, so 1 transactionid for the entire authentication process, not just each leg of our SAML request/response trips.

Reply:
Yeah, that's the idea based off of your comment in the associated issue:
> transaction id for user authentication (should generate a new value for this, unless we are upleveling in which case the previous transaction id should be used)

Does this make sense in that context?

Reply:
😄 I forgot that I wrote that!